pkg/route/validation/validation.go

```diff
@@ -342,6 +280,17 @@ func validateTLS(route *routev1.Route, fldPath *field.Path) field.ErrorList {
 		if len(tls.DestinationCACertificate) > 0 {
 			result = append(result, field.Invalid(fldPath.Child("destinationCACertificate"), "redacted destination ca certificate data", "edge termination does not support destination certificates"))
 		}
+
+		if opts.AllowExternalCertificates && tls.ExternalCertificate != nil {
+			if len(tls.Certificate) > 0 && len(tls.ExternalCertificate.Name) > 0 {
+				result = append(result, field.Invalid(fldPath.Child("externalCertificate"), tls.ExternalCertificate.Name, "cannot specify both tls.certificate and tls.externalCertificate"))
+				break
```

Comment:
The complementing else-if checks if external certificate  is present and attempts to validate external certificate. We can't have it just be an "else" since we need to check if external certificate is present. And without the break both if and else-if cases can be true. 

Attempting to validate external certificate when it's preconditions have failed isn't logically sound, hence the break.